ramp-database/rampdb/model/problem.py

```diff
@@ -32,87 +25,178 @@
 
 
 class Problem(Model):
+    """Problem table.
+
+    Parameters
+    ----------
+    name : str
+        The name of the problem.
+    path_ramp_kits : str
+        The path where the kits are located. It will corresponds to
+        the key `ramp_kits_dir` of the dictionary created with
+        :func:`ramputils.generate_ramp_config`.
+    path_ramp_data : str
+        The path where the data are located. It will corresponds to
+        the key `ramp_data_dir` of the dictionary created with
+        :func:`ramputils.generate_ramp_config`.
+    session : :class:`sqlalchemy.orm.Session`
+        The session to directly perform the operation on the database.
+
+    Attributes
+    ----------
+    id : int
+        The ID of the table row.
+    name : str
+        The name of the problem.
+    workflow_id : id
+        The ID of the associated workflow.
+    workflow : :class:`rampdb.model.Worflow`
+        The workflow instance.
+    path_ramp_kits : str
+        The path where the kits are located.
+    path_ramp_data : str
+        The path where the data are located.
+    events : list of :class:`rampdb.model.Event`
+        A back-reference to the event.
+    keywords : list of :class:`rampdb.model.ProblemKeyword`
+        A back-reference to the keywords associated with the problem.
+    """
     __tablename__ = 'problems'
 
     id = Column(Integer, primary_key=True)
     name = Column(String, nullable=False, unique=True)
 
-    workflow_id = Column(
-        Integer, ForeignKey('workflows.id'), nullable=False)
-    workflow = relationship(
-        'Workflow', backref=backref('problems'))
+    workflow_id = Column(Integer, ForeignKey('workflows.id'), nullable=False)
+    workflow = relationship('Workflow', backref=backref('problems'))
+
+    # XXX: big change in the database
+    path_ramp_kits = Column(String, nullable=False, unique=False)
+    path_ramp_data = Column(String, nullable=False, unique=False)
 
-    def __init__(self, name):
+    def __init__(self, name, path_ramp_kits, path_ramp_data, session=None):
         self.name = name
-        self.reset()
-        # to check if the module and all required fields are there
```

Comment:
I think this can be useful to keep (so that you already get a complaint on init time when there is something wrong with your Problem)